middle_end/flambda2/from_lambda/closure_conversion.ml

```diff
@@ -403,18 +403,21 @@ let close_c_call acc env ~loc ~let_bound_var
   let cost_metrics_of_body, free_names_of_body, acc, body =
     Acc.measure_cost_metrics acc ~f:(fun acc -> k acc None)
   in
+  (* CR layouts v2: I think this needs to change for primitives that return
+     #floats - presumably we should look at the second element of the 3-tuple
+     and do something different if it isn't value. *)
```

Comment:
Something I'd like to do once there are primitives that return `#float` is instead to add `Pbox_float` and `Punbox_float` lambda primitives, and in `translprim.ml`, convert those primitives to something like `box_float (prim (unbox_float arg)))`, completely removing the `prim_native_repr` part from lambda. This would simplify this part of the code quite a bit as well.

Reply:
That sounds nice!  But my instinct is the approach suggested here is the easiest thing for the very first `#float` release - what do you think?

Reply:
Yes, I think this approach is easier too, I just wanted to leave a comment so that we don't put too much effort now but do the minimum to handler this correctly, remembering to do this refactoring some time in the future, as it will simplify code in the future.

Reply:
Thanks - I've updated the comment to record this idea.  Do you have an opinion on @goldfirere suggestion below that we could eliminate the sort I've added to this tuple by storing it in the `Same_as_ocaml_repr` case?